api/validation_models.py

```
@@ -164,160 +157,119 @@ def validate_password_input(cls, v):
         v = validate_no_sql_injection(v)
         v = validate_no_xss(v)
         v = sanitize_html_input(v)
         return v
-    
-    @validator('parameters')
+
+    @validator("parameters")
     def validate_parameters(cls, v):
         """Validate parameters for security"""
         if not isinstance(v, dict):
             raise ValueError("Parameters must be a dictionary")
```

Ruff (0.14.1)

Prefer `TypeError` exception for invalid type

(TRY004)